File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/adapters/SearchRelationshipViewHolder.java
package org.dhis2.usescases.searchTrackEntity.adapters;

import android.graphics.drawable.Drawable;

import androidx.appcompat.content.res.AppCompatResources;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CircleCrop;

import org.dhis2.R;
import org.dhis2.databinding.ItemSearchRelationshipTrackedEntityBinding;
import org.dhis2.usescases.searchTrackEntity.SearchTEContractsModule;
import org.dhis2.utils.ObjectStyleUtils;
import org.hisp.dhis.android.core.trackedentity.TrackedEntityAttributeValue;

import java.io.File;
import java.util.List;

import static com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions.withCrossFade;

/**
 * Created by frodriguez on 13/5/2019.
 */

public class SearchRelationshipViewHolder extends RecyclerView.ViewHolder {

    private ItemSearchRelationshipTrackedEntityBinding binding;

    SearchRelationshipViewHolder(ItemSearchRelationshipTrackedEntityBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(SearchTEContractsModule.Presenter presenter, SearchTeiModel teiModel) {
        binding.setPresenter(presenter);

        setTEIData(teiModel.getAttributeValues());
        binding.executePendingBindings();
        itemView.setOnClickListener(view -> presenter.addRelationship(teiModel.getTei().uid(), teiModel.isOnline()));

        binding.trackedEntityImage.setBackground(AppCompatResources.getDrawable(itemView.getContext(), R.drawable.photo_temp_gray));
        File file = new File(teiModel.getProfilePicturePath());
        Drawable placeHolderId = ObjectStyleUtils.getIconResource(itemView.getContext(), teiModel.getDefaultTypeIcon(), R.drawable.photo_temp_gray);
        Glide.with(itemView.getContext())
                .load(file)
                .placeholder(placeHolderId)
                .error(placeHolderId)
                .transition(withCrossFade())
                .transform(new CircleCrop())
                .into(binding.trackedEntityImage);
    }

    private void setTEIData(List<TrackedEntityAttributeValue> trackedEntityAttributeValues) {
        binding.setAttribute(trackedEntityAttributeValues);
        binding.executePendingBindings();
    }


}

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/adapters/SearchRelationshipViewHolder.kt
package org.dhis2.usescases.searchTrackEntity.adapters

import android.view.View
import org.dhis2.commons.resources.ColorUtils
import org.dhis2.databinding.ItemSearchTrackedEntityBinding

class SearchRelationshipViewHolder(
    val binding: ItemSearchTrackedEntityBinding,
    val colorUtils: ColorUtils,
    private val onAddRelationship: (
        teiUid: String,
        relationshipTypeUid: String?,
        isOnline: Boolean,
    ) -> Unit,
) : BaseTeiViewHolder(binding, colorUtils) {

    override fun itemConfiguration() {
        binding.sortingFieldName.visibility = View.GONE
        binding.sortingFieldValue.visibility = View.GONE
        binding.syncState.visibility = View.GONE
    }

    override fun itemViewClick() {
        itemView.setOnClickListener {
            onAddRelationship(teiModel.tei.uid(), null, teiModel.isOnline)
        }
    }
}
